### ISSUE TITLE
SpikingCircus2 throwing error with multi-segment recordings

### ISSUE THREAD

**Issue by FrancescoNegri** [SpikeInterface/spikeinterface#2206]
Hi!
I was trying to run SpykingCircus2 with a multi-segment recording (3 segments for a total of ~18 mins) since, according to my understanding of the [documentation](https://spikeinterface.readthedocs.io/en/latest/modules/sorters.html#handling-multi-segment-recordings), only SpykingCircus2 and Tridesclous support multi-segment data.

However, the `run_sorter(...)` function throws this error:
```
{
	"name": "ValueError",
	"message": "This sorter spykingcircus2 does not handle multi-segment recordings, use si.concatenate_recordings(...)",
	"stack":  ...
}
```

Is such a behaviour expected?
Thank you in advance :)

**Comment by yger** [comment]
SC2 should work with multi segment, I'll investigate since I don't know where and why this message is raised...

**Comment by samuelgarcia** [comment]
Oups. This is a bug. I guess a bad assert somewhere. Pierre you fix this ?

**Comment by alejoe91** [comment]
Problem is here: https://github.com/SpikeInterface/spikeinterface/blob/main/src/spikeinterface/sorters/basesorter.py#L50

You need to set it to True in the SC2 class

**Comment by alejoe91** [comment]
Note that also TDC2 is missing it!

**Comment by yger** [comment]
This has been fixed in #2208
